src/connector/src/sink/iceberg/mod.rs

```diff
@@ -1446,6 +1502,7 @@ mod test {
             r#type: "upsert".to_owned(),
             force_append_only: false,
             primary_key: Some(vec!["v1".to_owned()]),
+            partition_by: Some(vec!["v1".to_owned(), "identity(v2)".to_owned()]),
```

Comment:
I think we should test all partition cases here, e.g
`v1, identity(v1), truncate(4,v2), bucket(5,v1), year(v3), month(v4), day(v5), hour(v6), void(v1)`